rotkehlchen/data_import/importers/bitmex.py

```diff
@@ -0,0 +1,166 @@
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class BitMEXWalletHistoryColnames(Enum):
+    transact_time = 'transactTime'
+    transact_type = 'transactType'
+    amount = 'amount'
+    fee = 'fee'
+    address = 'address'
+    transact_status = 'transactStatus'
+    wallet_balance = 'walletBalance'
+
+
+class BitMEXImporter(BaseExchangeImporter):
+    @staticmethod
+    def _consume_realised_pnl(
+            csv_row: dict[str, Any], timestamp_format: str = '%m/%d/%Y, %H:%M:%S %p',
+    ) -> tuple[MarginPosition, HistoryEvent]:
+        """
+        Use entries resulting from Realised PnL to generate
+        MarginPosition an object and HistoryEvent object.
+        May raise:
+        - KeyError
+        - DeserializationError
+        """
+        close_time = deserialize_timestamp_from_date(
+            date=csv_row['transactTime'],
+            formatstr=timestamp_format,
+            location='Bitmex Wallet History Import',
+        )
+        realised_pnl = AssetAmount(satoshis_to_btc(deserialize_asset_amount(csv_row['amount'])))
+        fee = deserialize_fee(csv_row['fee'])
+        notes = f"PnL from trade on {csv_row['address']}"
+
+        log.debug(
+            'Processing Bitmex Realised PnL',
+            timestamp=close_time,
+            profit_loss=realised_pnl,
+            fee=fee,
+            notes=notes,
+        )
+
+        event_identifier = f'BMEX_{hash_csv_row(csv_row)}'
+        usd_price = PriceHistorian().query_historical_price(A_BTC, A_USD, close_time)
+        abs_amount = realised_pnl.__abs__()  # pylint: disable=no-member
+        asset_balance = AssetBalance(A_BTC, Balance(abs_amount, usd_price * abs_amount))
+        if realised_pnl < 0:
+            history_event_subtype = HistoryEventSubType.SPEND
+        else:
+            history_event_subtype = HistoryEventSubType.RECEIVE
+
+        margin_position = MarginPosition(
+            location=Location.BITMEX,
+            open_time=None,
+            close_time=close_time,
+            profit_loss=realised_pnl,
+            pl_currency=asset_balance.asset,
+            fee=fee,
+            fee_currency=asset_balance.asset,
+            notes=notes,
+            link=f'Imported from BitMEX CSV file. Transact Type: {csv_row["transactType"]}',
+        )
+        history_event = HistoryEvent(
+            event_identifier=event_identifier,
+            sequence_index=0,
+            timestamp=ts_sec_to_ms(close_time),
+            location=Location.BITMEX,
+            asset=asset_balance.asset,
+            balance=asset_balance.balance,
+            notes=notes,
+            event_type=HistoryEventType.MARGIN,
+            event_subtype=history_event_subtype,
+        )
+
+        return margin_position, history_event
+
+    @staticmethod
+    def _consume_deposits_or_withdrawals(
+            csv_row: dict[str, Any], timestamp_format: str = '%m/%d/%Y, %H:%M:%S %p',
+    ) -> AssetMovement:
```

Comment:
Can you add a small comment with possible exceptions this can raise? It's the way we try to trace exception handling in the code